self-service-kiosk/src/app/views/menu/menu.js

```diff
@@ -0,0 +1,158 @@
+import {useDispatch, useSelector} from "react-redux";
+import React, {useRef, useState} from 'react';
+import {
+    getMenusAsync,
+    selectMenus
+} from "../../store/catalog-store.js";
+import {useEffect} from "react";
+import useModal from "../modal/useModal";
+import Modal from "../modal/modal";
+import "./menu.css"
+import {BottomSheet} from 'react-spring-bottom-sheet'
+import 'react-spring-bottom-sheet/dist/style.css'
+import {
+    addItemToOrderAsync,
+    removeItemToOrderAsync,
+    selectIdOrder,
+    selectItemsOrder,
+    startOrderAsync
+} from "../../store/order-store";
+import {useNavigate} from "react-router-dom";
+
+/**
+ * filter to get all distinct element in a list
+ */
+function onlyUnique(value, index, self) {
+    return self.indexOf(value) === index;
```

Comment:
??????????? j'arrive pas a comprendre

Reply:
Utilisée ligne 48 : menus.map(m => m.category).filter(onlyUnique);
index of retourne la première occurrence donc tu prends toujours uniquement le premier index qui contient la valeur. Ensuite tu gardes uniquement les onlyUnique qui sont ces premiers indexes et t'as qu'une fois la valeur.
cf : https://stackoverflow.com/questions/1960473/get-all-unique-values-in-a-javascript-array-remove-duplicates